src/azure/blobstore.rs

```diff
@@ -332,4 +344,73 @@ mod test {
 
         assert_eq!(b"barbell".to_vec(), result);
     }
+
+    #[tokio::test]
+    async fn get_cache_hit() -> Result<()> {
+        let server = MockServer::start().await;
+        let base_url = format!("{}/", server.uri());
+        let credentials =
+            AzureCredentials::new(&base_url, "account name", None, String::from("container"));
+        let container = HttpBlobContainer::new(&base_url, credentials.blob_container_name())?;
+
+        let body = b"hello".to_vec();
+        Mock::given(method("GET"))
+            .and(path("/container/foo/bar"))
+            .respond_with(ResponseTemplate::new(200).set_body_bytes(body.clone()))
+            .expect(1)
+            .mount(&server)
+            .await;
+
+        let result = container.get("foo/bar", &credentials).await?;
+        assert_eq!(result, body);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn get_cache_miss() -> Result<()> {
+        let server = MockServer::start().await;
+        let base_url = format!("{}/", server.uri());
+        let credentials =
+            AzureCredentials::new(&base_url, "account name", None, String::from("container"));
+        let container = HttpBlobContainer::new(&base_url, credentials.blob_container_name())?;
+
+        Mock::given(method("GET"))
+            .and(path("/container/foo/bar"))
+            .respond_with(ResponseTemplate::new(404))
+            .expect(1)
+            .mount(&server)
+            .await;
+
+        let result = container.get("foo/bar", &credentials).await;
+        match result {
+            Err(e) => match e.downcast::<BadHttpStatusError>() {
+                Ok(_) => Ok(()),
+                Err(e) => bail!("Result is not an Err(BadHttpStatusError): {}", e),
+            },
+            x => bail!("Result {:?} is not an Err(BadHttpStatusError)", x),
+        }
+    }
+
+    #[tokio::test]
+    async fn put() -> Result<()> {
+        let server = MockServer::start().await;
+        let base_url = format!("{}/", server.uri());
+        let credentials =
+            AzureCredentials::new(&base_url, "account name", None, String::from("container"));
+        let container = HttpBlobContainer::new(&base_url, credentials.blob_container_name())?;
+
+        let body = b"hello".to_vec();
+        Mock::given(method("PUT"))
+            .and(path("/container/foo/bar"))
+            .and(body_bytes(body.clone()))
+            .respond_with(ResponseTemplate::new(200))
+            .expect(1)
+            .mount(&server)
+            .await;
+
+        container.put("foo/bar", body, &credentials).await?;
```

Comment:
Is it possible to check that the data that `Mock` got matches `body`?

Reply:
Line 406 already does that.

Reply:
Ah, cool :+1: